### PR TITLE
Fix two Android specific Sync2 issues QueueSubmitBinarySemaphoreNotSignaled/Transfer stage validation

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
In commit e60d417b5 I didn't properly handle the different VUIDs
for devices that support timeline semaphores vs those that do not.

transfer stages were incorrect excluded from being valid on graphics and compute capable queues.

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2895,7 +2895,8 @@ struct SemaphoreSubmitState {
             (pSemaphore->scope == kSyncScopeInternal || internal_semaphores.count(semaphore))) {
             if (unsignaled_semaphores.count(semaphore) ||
                 (!(signaled_semaphores.count(semaphore)) && !(pSemaphore->signaled) && !core->SemaphoreWasSignaled(semaphore))) {
-                const auto &vuid = GetQueueSubmitVUID(loc, SubmitError::kSemCannotBeSignalled);
+                auto error = core->device_extensions.vk_khr_timeline_semaphore ? SubmitError::kTimelineCannotBeSignalled : SubmitError::kBinaryCannotBeSignalled;
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp:2898]: the name kTimelineCannotBeSignalled  is confusing.  It's not that a TimelineSemphore cannot be signaled... it's the "VK_KHR_timeline_semaphore is enabled *variant* of the "binary cannot be signaled" message. 

We should probably have a consistent naming scheme to clarify.  kBinaryCannotBeSignalledAltTimeline or hide the complexity in Get*VUID functions as needed somehow (pass in the extension structure?)

**Review by jzulauf-lunarg — CHANGES_REQUESTED** [review]
Enum name confusing/misleading.

**Review by jzulauf-lunarg — APPROVED** [review]
Can leave naming as it is for now...

**Pull request merged**